### PR TITLE
Add `cluster-autoscaler safe-to-evict` annotation to `controller` and `cainjector`

### PR DESCRIPTION
**Pull request by primeroz**
Towards https://github.com/giantswarm/vodafone/issues/617

This PR:

- Add `cluster-autoscaler safe-to-evict` annotation to `controller` and `cainjector`

#### Optional app

- [ ] fresh install
- [ ] upgrade from previous version

#### Pre-installed app

- [ ] fresh install during cluster creation
- [ ] upgrade from previous version in a pre-existing cluster

#### Other testing

- [ ] check reconciliation of existing resources after upgrading

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Added
+
+- Add `cluster-autoscaler safe-to-evict` annotation to `controller` and `cainjector`
```
Comment (ubergesundheit) [inline CHANGELOG.md:12]:
```suggestion
- Add `cluster-autoscaler safe-to-evict` annotation to `controller` and `cainjector`. ([#330](https://github.com/giantswarm/cert-manager-app/pull/330))
```

**Review by ubergesundheit — COMMENTED** [review]
Although this has the smell of introducing business logic into our charts, this looks okay if required immediately. 
Don't forget to request this as part of next platform releases as cert-manager is a default app.

IMO a better way would be to implement allowing users to specify annotations for the cainjector and controller deployments, then set this through other means (for example catalog default values)

**Comment by primeroz** [comment]
I will move the annotation to a field in the `values` so it can be moved to the catalog default values in case

**Comment by Gacko** [comment]
Well, this is required because we are running `cert-manager` in `kube-system`. If it was a non `kube-system` pod, autoscaler / Kubernetes would evict it without this annotation. For the same reason I removed it from `ingress-nginx` as I do not feel like it is really a `kube-system` app.

**Comment by primeroz** [comment]
> Well, this is required because we are running cert-manager in kube-system

100% to this

**Review by ubergesundheit — COMMENTED** [review]
Please update the changelog entry and update the values schema (See https://intranet.giantswarm.io/docs/dev-and-releng/app-developer-processes/helm-values-schema/)

**Comment by primeroz** [comment]
gonna update the schema now ( I thought we had a check for that ... )

**Comment by ubergesundheit** [comment]
@primeroz I took the liberty of updating schema and changelog for you.

The schema is a little wild, I admit :)

If you're cool with the changes, just merge :)

**Review by ubergesundheit — APPROVED** [review]

**Comment by primeroz** [comment]
@ubergesundheit 

> If you're cool with the changes, just merge :)

I am not just `cool with it` , i am **super thankful** :pray:

**Pull request merged**